### PR TITLE
Dev 1425 - agent to work with multiple streamsets

### PR DESCRIPTION
**Pull request by AntonZelenin**

### CODE REVIEW

File: agent/src/agent/cli/streamsets.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+def delete(url):
+    try:
+        streamsets_ = streamsets.repository.get_by_url(url)
+        pipelines = pipeline.repository.get_by_streamsets(streamsets_)
+        if _has_pipelines(pipelines):
+            # -1 because we don't count Monitoring
+            if not click.confirm(f'Streamsets with url {streamsets_.url} contains {len(pipelines) - 1} pipelines, all these pipelines will be DELETED, continue?'):
+                return
+            for pipeline_ in pipelines:
+                pipeline.repository.delete(pipeline_)
```
Comment (dariakharlan) [inline agent/src/agent/cli/streamsets.py:50]: Don't delete pipelines, move them to a different streamsts if available. I no streamsets available, don't delete anything

**Review by dariakharlan — CHANGES_REQUESTED** [review]
1. Please add a command for rebalancing existing pipelines between different stremastes. For example, when we already have a lot of pipelines in one streamsets instance, we are adding new one and we want to move there existing pipelines. Or we deleted a bunch of pipelines in one streamsets and we want to equally distribute the rest

File: agent/src/agent/pipeline/pipeline.py [hunk 2]
```diff
@@ -150,7 +150,7 @@ def timestamp_format(self) -> str:
 
     @property
     def values(self) -> list:
-        if self.source.type == source.TYPE_INFLUX:
+        if self.source.type == 'influx':
```
Comment (dariakharlan) [inline agent/src/agent/pipeline/pipeline.py:153]: why don't use constants anymore?

File: agent/src/agent/pipeline/pipeline.py [hunk 3]
```diff
@@ -283,6 +291,11 @@ def get_tags(self) -> dict:
         }
 
 
+class TestPipeline(Pipeline):
```
Comment (dariakharlan) [inline agent/src/agent/pipeline/pipeline.py:294]: what this class does?

File: agent/src/agent/streamsets/streamsets.py [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+    username = Column(String)
+    password = Column(String)
+
+    def __init__(self, url, username, password):
+        self.url = url
+        self.username = username
+        self.password = password
+
+
+class StreamSetsApiClient:
```
Comment (dariakharlan) [inline agent/src/agent/streamsets/streamsets.py:62]: so why did you move it from a separate file?

**Review by dariakharlan — CHANGES_REQUESTED** [review]

File: agent/src/agent/cli/__init__.py [hunk 5]
```diff
@@ -30,11 +32,28 @@ def agent(version):
         click.echo('Git commit: ' + __git_sha1__)
 
 
+def agent_entry_point():
+    try:
+        agent(standalone_mode=False)
+        db.session().commit()
+    except click.exceptions.Abort:
```
[the remark this answers is not in the document]
Comment (AntonZelenin) [reply agent/src/agent/cli/__init__.py:39]: indeed

File: agent/src/agent/cli/streamsets.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    click.echo(f'Streamsets `{url}` is successfully deleted from the agent')
+
+
+@click.command()
+def balance():
+    streamsets_ = streamsets.repository.get_all()
+    if len(streamsets_) == 1:
+        logger.info(f'You have only one streamsets instance, can\'t balance')
+        return
+    elif len(streamsets_) == 1:
```
[the remark this answers is not in the document]
Comment (AntonZelenin) [reply agent/src/agent/cli/streamsets.py:64]: yes, but I remembered that we won't have such situations, so I'll remove this condition

File: agent/src/agent/modules/logger.py [hunk 7]
```diff
@@ -1,14 +1,21 @@
 import logging
 import os
+import sys
 
 from logging.handlers import RotatingFileHandler
 
 
-def get_logger(name, level=logging.DEBUG):
+def get_logger(name, level=logging.DEBUG, stdout=False):
```
[the remark this answers is not in the document]
Comment (AntonZelenin) [reply agent/src/agent/modules/logger.py:8]: I don't see much difference between having separate functions and a flag, I can change it if you insist

File: agent/src/agent/pipeline/pipeline.py (hunk 2 above)
Comment (AntonZelenin) [reply agent/src/agent/pipeline/pipeline.py:153]: I removed source because of circular imports, I didn't try to fix it in this pr in order not to make it bigger, I'll leave a todo here, we can discuss this problem

File: agent/src/agent/pipeline/pipeline.py (hunk 3 above)
Comment (AntonZelenin) [reply agent/src/agent/pipeline/pipeline.py:294]: we choose an appropriate config loader depending on a Pipeline type, here https://github.com/anodot/daria/blob/1d5b8cf547491544ee319830fa951928fd24bacd/agent/src/agent/streamsets/manager.py#L109

File: agent/src/agent/streamsets/streamsets.py (hunk 4 above)
Comment (AntonZelenin) [reply agent/src/agent/streamsets/streamsets.py:62]: I didn't move it from a separate file, I moved that file entirely from modules directory, renamed it to streamsets and added StreamSets entity

**Review by dariakharlan — CHANGES_REQUESTED** [review]
1. Please add to the `pipeline list` command column with corresponding streamsets url
2. When creating a destination with cli - it shows that monitoring pipeline is created 
3. When trying to rebalance between two stremasets after creating 7 pipelines (3 in one, 4 in another) i got infinite loop:
```
root@agent:/usr/src/app# agent streamsets balance
2020-11-10 18:12:47,497 - INFO - Moving `k6` from `http://dc2:18630` to `http://dc2:18630`
2020-11-10 18:12:47,833 - INFO - Moved `k6` to `http://dc2:18630`
2020-11-10 18:12:47,846 - INFO - Moving `k6` from `http://dc2:18630` to `http://dc2:18630`
2020-11-10 18:12:48,133 - INFO - Moved `k6` to `http://dc2:18630`
2020-11-10 18:12:48,138 - INFO - Moving `k6` from `http://dc2:18630` to `http://dc2:18630`
```

**Review by dariakharlan — APPROVED** [review]

File: agent/src/agent/api/main.py [hunk 8]
```diff
@@ -38,4 +38,4 @@ def teardown_request_func(exception):
 
 @app.route('/version', methods=['GET'])
 def version():
-    return jsonify('Daria Agent version: ' + __version__)
+    return jsonify('Daria Agent version ' + __version__)
```
Comment (dariakharlan) [inline agent/src/agent/api/main.py:41]: why?
please check if it won't break release workflow step "check version"

Comment (AntonZelenin) [reply agent/src/agent/api/main.py:41]: I just checked how version is displayed in other apps and made the same, I also think that grammatically it's correct not to use a colon

**Review by dariakharlan — APPROVED** [review]

**Comment by AntonZelenin** [comment]
🎉
